course/viewbase.py

```diff
@@ -66,7 +67,13 @@ def get_resource_objects(self):
 
             # Apply course instance language.
             if self.instance.language:
-                translation.activate(self.instance.language)
+                lang = self.instance.language
+                if lang.startswith("|"):
+                    active = get_language()
+                    if "|" + active + "|" in lang:
+                        translation.activate(active)
```

Comment:
What happens if active language is not in the list?

Reply:
Fixed to fall back to the first language discovered in the course's settings so that the multilingual fields won't just disappear.